src/mscorlib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -138,10 +138,10 @@ public AdjustmentRule[] GetAdjustmentRules()
             {
                 var rule = _adjustmentRules[i];
                 var start = rule.DateStart.Kind == DateTimeKind.Utc ?
-                            new DateTime(TimeZoneInfo.ConvertTime(rule.DateStart, this).Ticks, DateTimeKind.Unspecified) :
+                            new DateTime(rule.DateStart.Ticks + _baseUtcOffset.Ticks, DateTimeKind.Unspecified) :
                             rule.DateStart;
                 var end = rule.DateEnd.Kind == DateTimeKind.Utc ?
-                            new DateTime(TimeZoneInfo.ConvertTime(rule.DateEnd, this).Ticks - 1, DateTimeKind.Unspecified) :
+                            new DateTime(rule.DateEnd.Ticks + _baseUtcOffset.Ticks + rule.DaylightDelta.Ticks, DateTimeKind.Unspecified) :
```

Comment:
yes. rule.DateEnd is UTC time which is mapping to the local time (with the daylight delta adjustment). 

for instance Pacific Time zone has _baseUtcOffset -08:00 hours and the daylight saving ends at 09:00 AM (which is equivalent to 2:00 AM when we having the daylight saving is on which is one hour delta) 

then the end time would be 09:00 - 08:00 + 01:00 which will give us 02:00 AM which is exactly the time the daylight saving end.